src/api/app/models/project.rb

```diff
@@ -168,11 +168,26 @@ def maintained_project_names
 
   # Check if the project has a path_element matching project and repository
   def has_distribution(project_name, repository)
-    project = Project.find_by(name: project_name)
-    return false unless project
+    has_local_distribution(repository) || has_remote_distribution(project_name, repository)
+  end
+
+  def has_remote_distribution(project_name, repository)
+    get_linked_repositories.remote.each do |linked_repository|
```

Comment:
Can you use `any?` instead `each + return` for this loop?
Something like:
```ruby
get_linked_repositories.remote.any? do |linked_repository|
  project_name.end_with?(linked_repository.remote_project_name) && linked_repository.name == repository
end
```

http://apidock.com/ruby/Enumerable/any%3F